make-portable.nix

```diff
@@ -0,0 +1,57 @@
+{ runCommand, patchelf, coreutils, binutils-unwrapped, find-libs }:
+
+targetPrefix: inputPackage:
+
+runCommand (inputPackage.pname + "-portable") {
+  inherit inputPackage targetPrefix;
+} ''
+  set -euo pipefail
+
+  cp -a "$inputPackage" "$out"
+  chmod -R u+w "$out"
+  mkdir -p "$out/lib"
+
+  processLib() {
+    lib="$1"
+    target="$2"
+
+    cp "$lib" "$target"
+    chmod u+w "$target"
+    if [ ! -z "$("${patchelf}/bin/patchelf" --print-rpath "$target")" ] ; then
+      "${patchelf}/bin/patchelf" --set-rpath "$targetPrefix/lib" "$target"
+    fi
+  }
+
+  addLib() {
+    lib="$1"
+
+    libname="$(${coreutils}/bin/basename "$lib")"
+    target="$out/lib/$libname"
+    if [ -e "$target" ] ; then
+      tmpfile="$("${coreutils}/bin/mktemp" "make-portable.XXXXXXXX")"
+      processLib "$lib" "$tmpfile"
+      diff "$target" "$tmpfile" >/dev/null
+      libsAreIdentical=$?
+      rm "$tmpfile"
+
+      if [ "$libsAreIdentical" -ne 0 ] ; then
+        2>&1 echo "error: multiple libraries named $libname are required, and they are not identical files"
+        exit 1
+      fi
+      echo "Library $lib already present - skipping"
+    else
+      echo "Adding library $lib"
+      processLib "$lib" "$target"
+    fi
+  }
+
+  for exe in $out/bin/* ; do
+    echo "Processing executable $exe"
+    for lib in $("${find-libs}/bin/find-libs" "$exe") ; do
+      addLib "$lib"
+    done
+    interpreter="$("${patchelf}/bin/patchelf" --print-interpreter "$exe")"
+    addLib "$interpreter"
+    "${patchelf}/bin/patchelf" --set-interpreter "$targetPrefix/lib/$("${coreutils}/bin/basename" "$interpreter")" --set-rpath "$targetPrefix/lib" "$exe"
```

Comment:
Unfortunately the interpreter can't use $ORIGIN, mentioned below, so that keeps you stuck with the /opt/chainweb absolute path. `/lib/ld-linux-${arch}.so.${ver}` is a safe bet for most Linux distros. Unfortunately that doesn't work on NixOS, though. See https://github.com/NixOS/nixpkgs/pull/69057/files#diff-aca5d4e8a2b0e3c27da452f0505a5a9dR224

Reply:
I tried using `/lib/ld-linux...` but it didn't seem to work for me - I got this issue: https://stackoverflow.com/questions/49245547/version-glibc-private-not-defined-in-file-ld-linux-x86-64-so-2

Reply:
Oh right - we would have to link against an older glibc to support older versions, which isn't very easy.